### PR TITLE
`Exam mode`: Do not display timer for unsubmitted exams

### PR DESCRIPTION
**Pull request by florian-glombik**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context
The timer is unnecessary, and it seems a bit broken to display `left time 0:0` when the exam and grace period is over.

### Description
Does not display the timer anymore after the exam is over, which was the case before for unsubmitted exams.

#### Exam Mode Testing
- 1 Instructor
- 1 Student

1. `Instructor`: Prepare an exam
2. `Student`: Verify the timer is still displayed during the exam
3. `Student`: Do not submit the exam
4. `Student`: wait until the grace period is over
5. `Student`: Try to open the exam summary
6. `Student`: See that the timer is still displayed, which would not be necessary

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### New
image

#### Old
image

### CODE REVIEW

File: src/test/javascript/spec/component/exam/participate/exam-participation-cover.component.spec.ts [hunk 1]
```diff
@@ -334,4 +335,24 @@ describe('ExamParticipationCoverComponent', () => {
...
+        expect(examTimerDebugElement).toBeFalsy();
+    });
+
+    it('should display timer during working time', () => {
+        jest.spyOn(component, 'studentFailedToSubmit', 'get').mockReturnValue(false);
+        component.startView = false;
+
+        fixture.detectChanges();
+
+        const examTimerDebugElement = fixture.debugElement.query(By.directive(ExamTimerComponent)); // Replace with the actual component class name
```
Comment (tobias-lippert) [inline src/test/javascript/spec/component/exam/participate/exam-participation-cover.component.spec.ts:355]: What do you want to express with the comment?

**Review by JanaNF — APPROVED** [review]
Manual tested on legacy ts2. Worked as expected.

**Review by jakubriegel — APPROVED** [review]
Works on ts1

**Review by vinceclifford — APPROVED** [review]
Tested on ts1

**Review by lennart-keller — APPROVED** [review]
Tested in testing session on ts 1

File: src/test/javascript/spec/component/exam/participate/exam-participation-cover.component.spec.ts (hunk 1 above)
Comment (florian-glombik) [reply src/test/javascript/spec/component/exam/participate/exam-participation-cover.component.spec.ts]: Nothing, I just forgot to remove it. 

Thanks for the heads up!

**Review by aplr — APPROVED** [review]
Tested in testing session, LGTM

**Review by lennart-keller — APPROVED** [review]
Reapprove

**Review by JohannesStoehr — APPROVED** [review]
Code

**Review by Gusti2010 — APPROVED** [review]
tested manually on ts2. works as expected

**Comment by MaximilianJG** [comment]
Works as expected. Tested together in a testing session with @Ceiridge on testing server 2. When the exercise has a due date in the future the other students (plagiarized) is not visible. However when the exercise does not have a due date the comparison is visible

**Comment by Ceiridge** [comment]
Tested with @MaximilianJG on test2. Everything worked as expected. (See his comment for screenshots)

**Comment by MaximilianJG** [comment]
Worked as expected. Tested in a exam testing session with @Ceiridge. Timer is displaying during exam and also after grace period. 

Screenshot 2023-11-15 at 17 56 46

Screenshot 2023-11-15 at 18 00 34

Also, timer if grace period is over is not shown anymore: 
Screenshot 2023-11-15 at 18 03 13

**Pull request merged**